src/main/java/picard/illumina/parser/TileMetricsUtil.java

```diff
@@ -229,7 +281,21 @@ private static File fileTileMetricsFile(File illuminaRunDirectory, Integer numCy
                         float[] phasingSlopeAndOffset = computeLinearFit(cycleNumWithData.toArray(new Float[0]), phasingMetrics.toArray(new Float[0]), phasingMetrics.size());
                         float[] prePhasingSlopeAndOffset = computeLinearFit(cycleNumWithData.toArray(new Float[0]), prephasingMetrics.toArray(new Float[0]), phasingMetrics.size());
                         TilePhasingValue value = new TilePhasingValue(tileTemplateRead, phasingSlopeAndOffset[0], prePhasingSlopeAndOffset[0]);
-                        tilePhasingValues.add(value);
+                        if (phasingValues.containsKey(laneNum)) {
```

Comment:
I believe all this logic can be replaced with two calls to `computeIfAbsent`, e.g.
```
phasingValues.computeIfAbsent(laneNum, k -> new HashMap<>()).computeIfAbsent(tileNum, k -> new ArrayList<>()).add(value);
```

Reply:
Ah! I was hoping there might be something like this

Reply:
It looks like the code above could also benefit from this API
```
int laneNumber = laneTileCode.getLaneNumber();
phasing.computeIfAbsent(tileNumber, k -> new HashMap<>()).computeIfAbsent(laneNumber, k -> new ArrayList<>()).add(phasingMetrics.phasingWeight);
prePhasing.computeIfAbsent(tileNumber, k -> new HashMap<>()).computeIfAbsent(laneNumber, k -> new ArrayList<>()).add(phasingMetrics.prephasingWeight);
```